### PR TITLE
[clang][analyzer][doc] Migrate ClangSA www FAQ section

### PR DESCRIPTION
**Pull request by gamesh411**
The ClangSA documentation lives in RST format, and the FAQ section of the old webpage is also migrated from HTML with this change.

### CODE REVIEW

File: clang/docs/analyzer/user-docs/FAQ.rst [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+.. code-block:: none
+
+   Value stored to 'x' is never read
+
+You can use the ``(void)x;`` idiom to acknowledge that there is a dead store in your code but you do not want it to be reported in the future.
+
+Unused Instance Variable
+------------------------
+
+Q: How do I tell the static analyzer that I don't care about a specific unused instance variable in Objective-C?
```
Comment (NagyDonat) [inline clang/docs/analyzer/user-docs/FAQ.rst:54]: Perhaps extend this to advice about analogous issues with C/C++.

File: clang/docs/analyzer/user-docs/FAQ.rst [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+.. code-block:: none
+
+   Instance variable 'commonName' in class 'HappyBird' is never used by the methods in its @implementation
+
+You can add ``__attribute__((unused))`` to the instance variable declaration to suppress the warning.
+
+Unlocalized String
+------------------
+
+Q: How do I tell the static analyzer that I don't care about a specific unlocalized string?
```
Comment (NagyDonat) [inline clang/docs/analyzer/user-docs/FAQ.rst:67]: This is oddly specific advice, I don't see why is it in the "frequently" asked questions. (Although I'm not native to ObjCLand, maybe this is a common issue there...)

File: clang/docs/analyzer/user-docs/FAQ.rst [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+.. code-block:: objc
+
+   return (id _Nonnull)nil;
+
+Note that this cast does not affect code generation.
+
+Ensuring Loop Body Execution
+----------------------------
+
+Q: The analyzer assumes that a loop body is never entered. How can I tell it that the loop body will be entered at least once?
```
Comment (NagyDonat) [inline clang/docs/analyzer/user-docs/FAQ.rst:144]: I'd say that this is a bug (unwanted counter-intuitive behavior) and I hope that a generalized version of https://github.com/llvm/llvm-project/pull/109804 will eventually eliminate this question.

File: clang/www/analyzer/faq.html [hunk 4]
```diff
@@ -18,242 +17,11 @@
 <div id="content">
 
 <h1>FAQ and How to Deal with Common False Positives</h1>
+<p style="color:red; font-size:200%">This page is deprecated and will be removed in release 21.0</p>
+<a href="https://clang.llvm.org/docs/analyzer/user-docs/FAQ.html">The new site</a>
```
Comment (NagyDonat) [inline clang/www/analyzer/faq.html:21]:
```suggestion
<p>Its content was migrated to <a href="https://clang.llvm.org/docs/analyzer/user-docs/FAQ.html">the regular LLVM documentation</a>.</p>
```

**Review by NagyDonat — COMMENTED** [review]
Overall LGTM, with some minor bikeshedding on the redirect page.

I also added several inline comments about the _content_ of the page, but the content changes should be left for a separate follow-up commit to separate migrating the text and changing it.

**Comment by steakhal** [comment]
I more than welcome migrating from raw html to rst. - or basically to any other format.

**Pull request merged**